pipeline/recorder/src/main/scala/uk/ac/wellcome/platform/recorder/services/RecorderWorkerService.scala

```diff
@@ -1,53 +1,57 @@
 package uk.ac.wellcome.platform.recorder.services
 
+import scala.util.{Failure, Success, Try}
+import scala.concurrent.Future
 import akka.Done
+
 import uk.ac.wellcome.json.JsonUtil._
-import uk.ac.wellcome.messaging.message.{
-  MessageNotification,
-  MessageStream,
-  RemoteNotification
-}
-import uk.ac.wellcome.messaging.sns.SNSWriter
 import uk.ac.wellcome.models.work.internal.TransformedBaseWork
-import uk.ac.wellcome.storage.ObjectStore
-import uk.ac.wellcome.storage.dynamo._
-import uk.ac.wellcome.storage.vhs.{
-  EmptyMetadata,
-  VHSIndexEntry,
-  VersionedHybridStore
-}
 import uk.ac.wellcome.typesafe.Runnable
 
-import scala.concurrent.{ExecutionContext, Future}
+import uk.ac.wellcome.bigmessaging.typesafe.{EmptyMetadata, GetLocation}
+import uk.ac.wellcome.messaging.MessageSender
+import uk.ac.wellcome.bigmessaging.message.MessageStream
 
-class RecorderWorkerService(
-  versionedHybridStore: VersionedHybridStore[TransformedBaseWork,
-                                             EmptyMetadata,
-                                             ObjectStore[TransformedBaseWork]],
+import uk.ac.wellcome.storage.store.{HybridStoreEntry, VersionedStore}
+import uk.ac.wellcome.storage.{Identified, Version}
+
+class RecorderWorkerService[MsgDestination](
+  store: VersionedStore[
+    String,
+    Int,
+    HybridStoreEntry[TransformedBaseWork, EmptyMetadata]] with GetLocation,
   messageStream: MessageStream[TransformedBaseWork],
-  snsWriter: SNSWriter)(implicit ec: ExecutionContext)
+  msgSender: MessageSender[MsgDestination])
     extends Runnable {
 
   def run(): Future[Done] =
     messageStream.foreach(this.getClass.getSimpleName, processMessage)
 
   private def processMessage(work: TransformedBaseWork): Future[Unit] =
-    for {
-      vhsEntry <- storeInVhs(work)
-      _ <- snsWriter.writeMessage[MessageNotification](
-        message = RemoteNotification(vhsEntry.hybridRecord.location),
-        subject = s"Sent from ${this.getClass.getSimpleName}")
-    } yield ()
-
-  private def storeInVhs(
-    work: TransformedBaseWork): Future[VHSIndexEntry[EmptyMetadata]] =
-    versionedHybridStore.updateRecord(work.sourceIdentifier.toString)(
-      (work, EmptyMetadata()))(
-      (existingWork, existingMetadata) =>
-        if (existingWork.version > work.version) {
-          (existingWork, existingMetadata)
-        } else {
-          (work, EmptyMetadata())
+    Future.fromTry {
+      for {
+        key <- storeWork(work)
+        location <- store.getLocation(key)
+        _ <- msgSender.sendT(location)
+      } yield ()
+    }
```

Comment:
In the storage service we've exposed this method publicly, so we can call it directly in tests and write tests that avoid the need to set up an SQS queue.

No need to change now, but might be worth considering for the catalogue at some point – and then change this to return a `Try`, and move the `Future.fromTry` into the `messageStream.foreach`.